src/lib/operators/export_binary.cpp

```diff
@@ -164,6 +174,56 @@ void ExportBinary::_write_chunk(const std::shared_ptr<const Table>& table, std::
   }
 }
 
+void ExportBinary::_write_partitioning_header(const std::shared_ptr<const Table>& table, std::ofstream& ofstream) {
+  const auto partition_schema = table->get_partition_schema();
+  _export_value(ofstream, partition_schema_type_to_uint.left.at(partition_schema->get_type()));
+  _export_value(ofstream, partition_schema->partition_count());
+
+  switch (partition_schema->get_type()) {
+    case PartitionSchemaType::Hash: {
+      const auto hash_schema = std::dynamic_pointer_cast<const HashPartitionSchema>(partition_schema);
+      _export_value(ofstream, static_cast<ColumnID>(hash_schema->get_column_id()));
+      break;
+    }
+    case PartitionSchemaType::Range: {
+      const auto range_schema = std::dynamic_pointer_cast<const RangePartitionSchema>(partition_schema);
+      const auto bound_type_string = data_type_to_string.left.at(range_schema->get_bound_type());
+      _export_value(ofstream, static_cast<ColumnID>(range_schema->get_column_id()));
+      _export_values(ofstream, std::vector<std::string>{bound_type_string});
+      resolve_data_type(range_schema->get_bound_type(), [&](auto type) {
+        using VectorDataType = typename decltype(type)::type;
+        std::vector<VectorDataType> typed_bounds;
+        typed_bounds.reserve(range_schema->get_bounds().size());
+        for (const auto& bound : range_schema->get_bounds()) {
+          typed_bounds.emplace_back(get<VectorDataType>(bound));
+        }
+        _export_values(ofstream, typed_bounds);
+      });
+      break;
+    }
+    case PartitionSchemaType::RoundRobin:
+    case PartitionSchemaType::Null: {
+      break;
```

Comment:
If these types do not require any additional information, please leave a comment so that people are not surprised about the empty block.